api/src/wfl/api/handlers.clj

```diff
@@ -90,41 +90,39 @@
   "Return the workflows managed by the workload."
   [request]
   (log/info (select-keys request [:request-method :uri :parameters]))
-  (let [uuid (get-in request [:path-params :uuid])
-        status (get-in request [:parameters :query :status])]
+  (let [uuid    (get-in request [:path-params :uuid])
+        filters (-> request
+                    (get-in [:parameters :query])
+                    (select-keys [:submission :status]))]
     (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
       (->> (let [workload (workloads/load-workload-for-uuid tx uuid)]
-             (if status (workloads/workflows-by-status tx workload status) (workloads/workflows tx workload)))
+             (if (empty? filters)
+               (workloads/workflows tx workload)
+               (workloads/workflows-by-filters tx workload filters)))
            (mapv util/to-edn)
            succeed))))
 
 ;; Visible for testing
-(def retry-unsupported-status-error-message
-  "Retry unsupported for requested status.")
 (def retry-no-workflows-error-message
-  "No workflows to retry for requested status.")
+  "No workflows to retry for workload and requested filters.")
 
 (defn post-retry
   "Retry the workflows identified in `request`."
   [request]
   (log/info (select-keys request [:request-method :uri :parameters]))
   (let [uuid       (get-in request [:path-params :uuid])
-        status     (get-in request [:body-params :status])
-        supported? (cromwell/retry-status? status)]
-    (when-not supported?
-      (throw (UserException. retry-unsupported-status-error-message
-                             {:workload           uuid
-                              :supported-statuses cromwell/retry-status?
-                              :requested-status   status
-                              :status             400})))
+        filters    (-> request
+                       :body-params
+                       (select-keys [:submission :status]))]
     (->> (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
            (let [workload  (workloads/load-workload-for-uuid tx uuid)
-                 workflows (workloads/workflows-by-status tx workload status)]
+                 _         (workloads/throw-if-invalid-retry-filters workload filters)
```

Comment:
The validation we wish to perform on the request's workflow filters depend on the type of workload on which we're operating.

Initially, we only have the workload's UUID.  Once we've loaded the workload, we can then check that the filters specified are sufficient.

As such, I moved the check on workflow status being supported to `executor/terra-executor-throw-if-invalid-retry-filters`.